RHDEVDOCS-4068 - Logging 5.5 Release Notes

libander opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OSDOCS-<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch. --->

Version(s): 4.10+
<!--- Specify the version or versions of OpenShift your PR applies to.

Examples:
  * PR applies to all versions after a specific version (e.g. 4.8): 4.8+
  * PR applies to the in-development version (e.g. 4.11) and future versions: 4.11+
  * PR applies only to a specific single version (e.g. 4.10): 4.10
  * PR applies to multiple specific versions (e.g. 4.6-4.8): 4.6, 4.7, 4.8 --->

Issue: https://issues.redhat.com/browse/RHDEVDOCS-4068
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

Link to docs preview: https://drive.google.com/file/d/1mxVZOFLLlHTyhsHbVzw7gRs2A8mWB4QV/view?usp=sharing
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build.


Automatic preview functionality is currently not available.
  * OpenShift documentation team members (core and aligned) must include a link to a locally generated preview for PRs that update the rendered build in any way.
  * External contributors can request a generated preview from the OpenShift documentation team. --->
Reviewers:
@periklis
@jcantrill
@xperimental
@anpingli
@kabirbhartiRH 
@QiaolingTang

Peer review note: Errata link & CVEs to be added when Errata is generated.

jcantrill commented on modules/cluster-logging-rn-5.5.adoc:
```diff
@@ -0,0 +1,53 @@
+JSON formatting of logs varies by application. This feature should be used sparingly to create indices to separate logs with incompatible JSON formats, as creating too many indices will impact performance. To separate logs from different namespaces, or applications with compatible JSON formats, use queries.
+====
+
+* With this update, you can filter logs with Elasticsearch outputs using the Kubernetes common labels `app_kubernetes_io/component`,`app_kubernetes_io/managed-by`,`app_kubernetes_io/part-of`, and `app_kubernetes_io/version`. Non-Elasticsearch output types can use all labels included in `kubernetes.labels`. (link:https://issues.redhat.com/browse/LOG-2388[LOG-2388])
```
We modified to remove the underscores.  You can replace them with dots

jcantrill commented on modules/cluster-logging-rn-5.5.adoc:
```diff
@@ -0,0 +1,53 @@
+
+[id="openshift-logging-5-5-0-bug-fixes"]
+== Bug fixes
+* Before this update, clusters configured to forward logs to Amazon CloudWatch wrote rejected log files to temporary storage, causing cluster instability over time. With this update, chunk backup for CloudWatch has been disabled, resolving the issue. (link:https://issues.redhat.com/browse/LOG-2746[LOG-2746])
```
This was updated to disable chunk backup for any storage

jcantrill commented:
@libander general question about release notes for new Y-stream releases:  do we include bug text for all bugs or just the ones that are newly fixed in the Y-stream?  Is it possible we have RN for bugs that we fixed in earlier versions (e.g. cloudwatch chunk backup)?  Is it possible we (devs) are not setting the proper label or fields on JIRA issues to this does not happen?

libander commented:
> @libander general question about release notes for new Y-stream releases: do we include bug text for all bugs or just the ones that are newly fixed in the Y-stream? Is it possible we have RN for bugs that we fixed in earlier versions (e.g. cloudwatch chunk backup)? Is it possible we (devs) are not setting the proper label or fields on JIRA issues to this does not happen?

Good question. I noticed that too, I suppose logically if it was fixed in an earlier release then including it on a new Y stream release doesn't make much sense. Just mark Y stream iterations of bugs resolved in prior Z streams as no-doc. I'll remove the bugs here that have prior version fixes.

libander commented:
@jcantrill - Updated, removed bug fixes that exist in prior release notes. 
Updated preview: https://drive.google.com/file/d/1mxVZOFLLlHTyhsHbVzw7gRs2A8mWB4QV/view?usp=sharing

rolfedh commented on modules/cluster-logging-rn-5.5.adoc:
```diff
@@ -0,0 +1,35 @@
+//cluster-logging-release-notes
+[id="cluster-logging-release-notes-5-5-0"]
+= Logging 5.5
+The following advisories are available for logging 5.5:
```
```suggestion
The following advisories are available for Logging 5.5:
```

rolfedh commented on modules/cluster-logging-rn-5.5.adoc:
```diff
@@ -0,0 +1,35 @@
+
+[id="openshift-logging-5-5-0-enhancements"]
+== Enhancements
+* With this update, you can forward structured logs from different containers within the same pod to different indices. To use this feature, you must configure the pipeline with multi-container support, and annotate the pods. (link:https://issues.redhat.com/browse/LOG-1296[LOG-1296])
```
```suggestion
* With this update, you can forward structured logs from different containers within the same pod to different indices. To use this feature, you must configure the pipeline with multi-container support and annotate the pods. (link:https://issues.redhat.com/browse/LOG-1296[LOG-1296])
```

rolfedh commented on modules/cluster-logging-rn-5.5.adoc:
```diff
@@ -0,0 +1,35 @@
+
+[IMPORTANT]
+====
+JSON formatting of logs varies by application. This feature should be used sparingly to create indices to separate logs with incompatible JSON formats, as creating too many indices will impact performance. To separate logs from different namespaces, or applications with compatible JSON formats, use queries.
```
```suggestion
JSON formatting of logs varies by application. Because creating too many indices impacts performance, limit your use of this feature to creating indices for logs that have incompatible JSON formats. Use queries to separate logs from different namespaces, or applications with compatible JSON formats.
```

rolfedh commented on modules/cluster-logging-rn-5.5.adoc:
```diff
@@ -0,0 +1,35 @@
+JSON formatting of logs varies by application. This feature should be used sparingly to create indices to separate logs with incompatible JSON formats, as creating too many indices will impact performance. To separate logs from different namespaces, or applications with compatible JSON formats, use queries.
+====
+
+* With this update, you can filter logs with Elasticsearch outputs using the Kubernetes common labels `app.kubernetes.io/component`,`app.kubernetes.io/managed-by`,`app.kubernetes.io/part-of`, and `app.kubernetes.io/version`. Non-Elasticsearch output types can use all labels included in `kubernetes.labels`. (link:https://issues.redhat.com/browse/LOG-2388[LOG-2388])
```
```suggestion
* With this update, you can filter logs with Elasticsearch outputs by using the Kubernetes common labels, `app.kubernetes.io/component`, `app.kubernetes.io/managed-by`, `app.kubernetes.io/part-of`, and `app.kubernetes.io/version`. Non-Elasticsearch output types can use all labels included in `kubernetes.labels`. (link:https://issues.redhat.com/browse/LOG-2388[LOG-2388])
```

rolfedh commented on modules/cluster-logging-rn-5.5.adoc:
```diff
@@ -0,0 +1,35 @@
+
+* Before this update the metrics server for the collector attempted to bind to the address using a value exposed by an environment value. This change modifies the configuration to bind to any available interface. (link:https://issues.redhat.com/browse/LOG-2675[LOG-2675])
+
+* Before this update, the Operator was using versions of some APIs which are pending removal in future versions of {product-title}. This change moves dependencies to the supported API versions. (link:https://issues.redhat.com/browse/LOG-2656[LOG-2656])
```
```suggestion
* Before this update, the Operator was using versions of some APIs that are deprecated and planned for removal in future versions of {product-title}. This update moves dependencies to the supported API versions. (link:https://issues.redhat.com/browse/LOG-2656[LOG-2656])
```

rolfedh commented on modules/cluster-logging-rn-5.5.adoc:
```diff
@@ -0,0 +1,35 @@
+
+* Before this update, the Operator was using versions of some APIs which are pending removal in future versions of {product-title}. This change moves dependencies to the supported API versions. (link:https://issues.redhat.com/browse/LOG-2656[LOG-2656])
+
+* Before this update, multiple `ClusterLogForwarder` pipelines configured for multiline error detection would cause the collector to go into a `crashloopbackoff` error state.  This changes fixes the issue where multiple configuration sections had the same unique id. (link:https://issues.redhat.com/browse/LOG-2241[LOG-2241])
```
```suggestion
* Before this update, multiple `ClusterLogForwarder` pipelines configured for multiline error detection caused the collector to go into a `crashloopbackoff` error state. This update fixes the issue where multiple configuration sections had the same unique ID. (link:https://issues.redhat.com/browse/LOG-2241[LOG-2241])
```

rolfedh commented on modules/cluster-logging-rn-5.5.adoc:
```diff
@@ -0,0 +1,35 @@
+
+* Before this update, multiple `ClusterLogForwarder` pipelines configured for multiline error detection would cause the collector to go into a `crashloopbackoff` error state.  This changes fixes the issue where multiple configuration sections had the same unique id. (link:https://issues.redhat.com/browse/LOG-2241[LOG-2241])
+
+* Before this update, the collector could not save non UTF-8 symbols to the Elasticsearch storage logs. With this update the collector encodes non UTF-8 symbols, resolving the issue. (link:https://issues.redhat.com/browse/LOG-2203[LOG-2203])
```
```suggestion
* Before this update, the collector did not save non-UTF-8 symbols to the Elasticsearch storage logs. With this update, the collector encodes non-UTF-8 symbols, resolving the issue. (link:https://issues.redhat.com/browse/LOG-2203[LOG-2203])
```

rolfedh commented on modules/cluster-logging-rn-5.5.adoc:
```diff
@@ -0,0 +1,35 @@
+== Bug fixes
+* Before this update, clusters configured to forward logs to Amazon CloudWatch wrote rejected log files to temporary storage, causing cluster instability over time. With this update, chunk backup for all storage options has been disabled, resolving the issue. (link:https://issues.redhat.com/browse/LOG-2746[LOG-2746])
+
+* Before this update the metrics server for the collector attempted to bind to the address using a value exposed by an environment value. This change modifies the configuration to bind to any available interface. (link:https://issues.redhat.com/browse/LOG-2675[LOG-2675])
```
```suggestion
* Before this update, the metrics server for the collector attempted to bind to the address using a value exposed by an environment value. This update modifies the configuration to bind to any available interface. (link:https://issues.redhat.com/browse/LOG-2675[LOG-2675])
```

rolfedh commented on modules/cluster-logging-rn-5.5.adoc:
```diff
@@ -0,0 +1,35 @@
+
+* Before this update, the collector could not save non UTF-8 symbols to the Elasticsearch storage logs. With this update the collector encodes non UTF-8 symbols, resolving the issue. (link:https://issues.redhat.com/browse/LOG-2203[LOG-2203])
+
+* Before this update, non latin characters displayed incorrectly in Kibana. With this update Kibana displays all valid UTF-8 symbols correctly. (link:https://issues.redhat.com/browse/LOG-2784[LOG-2784])
```
```suggestion
* Before this update, non-latin characters displayed incorrectly in Kibana. With this update, Kibana displays all valid UTF-8 symbols correctly. (link:https://issues.redhat.com/browse/LOG-2784[LOG-2784])
```

rolfedh reviewed (approved):
Pre-approved to expedite. Nice work.

jcantrill commented on modules/cluster-logging-rn-5.5.adoc:
```diff
@@ -0,0 +1,38 @@
+
+* Before this update, the Operator was using versions of some APIs that are deprecated and planned for removal in future versions of {product-title}. This update moves dependencies to the supported API versions. (link:https://issues.redhat.com/browse/LOG-2656[LOG-2656])
+
+Before this update, the Operator was using versions of some APIs that are deprecated and planned for removal in future versions of {product-title}. This update moves dependencies to the supported API versions. (link:https://issues.redhat.com/browse/LOG-2656[LOG-2656])
```
This is a duplicate of the prior bullet

jcantrill commented:
lgtm

jcantrill commented:
/lgtm

kabirbhartiRH commented:
/lgtm

anpingli commented:
/lgtm

libander commented:
https://drive.google.com/file/d/1JqPbFD9Jb9NMyQ83_0r0nbGhYgQXr81Q/view?usp=sharing

JStickler commented:
/cherry-pick enterprise-4.10

JStickler commented:
/cherry-pick enterprise-4.11

JStickler commented:
/cherry-pick enterprise-4.12
